From 726aef1f9d58eecfc1add8ca0552779658713d0a Mon Sep 17 00:00:00 2001
From: Graham Christensen <graham@grahamc.com>
Date: Wed, 24 Apr 2019 19:51:17 -0400
Subject: [PATCH] 24 -> 32

---
 sway/input/seat.c | 2 +-
 sway/server.c     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sway/input/seat.c b/sway/input/seat.c
index be523539..3017899b 100644
--- a/sway/input/seat.c
+++ b/sway/input/seat.c
@@ -703,7 +703,7 @@ void seat_configure_xcursor(struct sway_seat *seat) {
 
 	if (!seat->cursor->xcursor_manager) {
 		seat->cursor->xcursor_manager =
-			wlr_xcursor_manager_create(cursor_theme, 24);
+			wlr_xcursor_manager_create(cursor_theme, 32);
 		if (sway_assert(seat->cursor->xcursor_manager,
 					"Cannot create XCursor manager for theme")) {
 			return;
diff --git a/sway/server.c b/sway/server.c
index cbb5936e..6c17b0e7 100644
--- a/sway/server.c
+++ b/sway/server.c
@@ -160,7 +160,7 @@ void server_fini(struct sway_server *server) {
 
 bool server_start(struct sway_server *server) {
 	// TODO: configurable cursor theme and size
-	int cursor_size = 24;
+	int cursor_size = 32;
 	const char *cursor_theme = NULL;
 
 	char cursor_size_fmt[16];
-- 
2.19.2

